_pytest/mark.py

```diff
@@ -329,30 +336,40 @@ def __call__(self, *args, **kwargs):
             is_class = inspect.isclass(func)
             if len(args) == 1 and (istestfunc(func) or is_class):
                 if is_class:
-                    if hasattr(func, 'pytestmark'):
-                        mark_list = func.pytestmark
-                        if not isinstance(mark_list, list):
-                            mark_list = [mark_list]
-                        # always work on a copy to avoid updating pytestmark
-                        # from a superclass by accident
-                        mark_list = mark_list + [self]
-                        func.pytestmark = mark_list
-                    else:
-                        func.pytestmark = [self]
+                    apply_mark(func, self.mark)
                 else:
-                    holder = getattr(func, self.name, None)
-                    if holder is None:
-                        holder = MarkInfo(self.mark)
-                        setattr(func, self.name, holder)
-                    else:
-                        holder.add_mark(self.mark)
+                    apply_legacy_mark(func, self.mark)
+                    apply_mark(func, self.mark)
                 return func
 
         mark = Mark(self.name, args, kwargs)
         return self.__class__(self.mark.combined_with(mark))
 
 
+def apply_mark(obj, mark):
+    assert isinstance(mark, Mark), mark
+    """applies a marker to an object,
```

Comment:
pytestmark will stay, no need to change it

im against a get_pristine_marks function, - the marks api people use to get/interact with marks sould be an new object on item that we will provide

after all marks in and from parameterization are to be considered as well and those are a pain to get